### PR TITLE
Sidechains integration step4 cert sc id

### PR DESCRIPTION
**Pull request by alsala**
the sc id has been added to cert prover/verifier/hash. A suited mc crypto lib branch is pointed to.

### CODE REVIEW

File: src/primitives/certificate.cpp [hunk 1]
```diff
@@ -338,14 +338,18 @@ CFieldElement CScCertificate::GetDataHash(const Sidechain::ScFixedParameters& sc
     if (bt_list_len == 0)
         bt_list_ptr = nullptr;
 
+    wrappedFieldPtr sptrScId = CFieldElement(input.scId).GetFieldElement();
+    field_t* scidFe = sptrScId.get();
+
     wrappedFieldPtr   sptrConst  = input.constant.GetFieldElement();
     wrappedFieldPtr   sptrCum    = input.endEpochCumScTxCommTreeRoot.GetFieldElement();
     wrappedScProofPtr sptrProof  = input.certProof.GetProofPtr();
     wrappedScVkeyPtr  sptrCertVk = input.CertVk.GetVKeyPtr();
```
Comment (i-Alex) [inline src/primitives/certificate.cpp:347]: the 4 items above are not used. Do we need to keep them?

**Review by i-Alex — APPROVED** [review]

File: src/primitives/certificate.cpp (hunk 1 above)
Comment (alsala) [reply src/primitives/certificate.cpp]: Ok, removed; also added log trace with return code value

**Review by i-Alex — APPROVED** [review]

File: qa/rpc-tests/run_sc_tests.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+rm /tmp/result.log
+
+for i in sbh_rpc_cmds.py rawtransactions.py getblocktemplate_proposals.py listtransactions.py ws_messages.py sc_*py
+do
+	echo "$i ---------------------------" | tee -a /tmp/result.log
+	time ./$i | tee -a /tmp/result.log
```
[the remark this answers is not in the document]
Comment (alsala) [reply qa/rpc-tests/run_sc_tests.sh]: Done, also added stdbuf for having unbuffered output on console

File: qa/rpc-tests/run_sc_tests.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
```
[the remark this answers is not in the document]
Comment (alsala) [reply qa/rpc-tests/run_sc_tests.sh:2]: Done

**Review by ptagl — APPROVED** [review]

**Comment by cronicc** [comment]
Conflicts with #130

**Review by albertog78 — APPROVED** [review]

**Pull request merged**